addon/components/menu.js

```diff
@@ -104,25 +108,25 @@ export default class Menu extends Component {
 
   @action
   registerItem(item) {
```

Comment:
I see what you mean about the double render error here. This code is kinda problematic (likely copied from the original implementation!)

Do you know if the list still updates of items are added post-render?

My concern is that we've last a bit of reactivity, and I wonder if it'd be better to instead detach from the tracking fram during render via an `await Promise.resolve()` in each of these methods.

Thoughts?

Reply:
I won't pretend to understand the mechanics (feel free to explain when you have time) but what you proposed works and the refactoring is done. 👍🏻